pkgs/by-name/op/opencl-clang/llvm/default.nix

```diff
@@ -0,0 +1,115 @@
+{
+  newScope,
+  lib,
+  fetchFromGitHub,
+  wrapCCWith,
+  buildPackages,
+  buildLlvmTools, # tools, but from the previous stage, for cross
+  targetLlvmLibraries, # libraries, but from the next stage, for cross
+  sha256,
+  version,
+  ...
+}@args:
+
+let
+  metadata = rec {
+    inherit
+      (import ./common-let.nix {
+        inherit
+          lib
+          fetchFromGitHub
+          sha256
+          version
+          ;
+      })
+      llvm_meta
+      monorepoSrc
+      ;
+    src = monorepoSrc;
+  };
+
+  tools = lib.makeExtensible (
+    tools:
+    let
+      callPackage = newScope (tools // args // metadata);
+      mkExtraBuildCommands0 =
+        cc:
+        ''
+          rsrc="$out/resource-root"
+          mkdir "$rsrc"
+          echo "-resource-dir=$rsrc" >> $out/nix-support/cc-cflags
+        ''
+        + ''
+          ln -s "${lib.getLib cc}/lib/clang/${version}/include" "$rsrc"
+        '';
+      mkExtraBuildCommands =
+        cc:
+        mkExtraBuildCommands0 cc
+        + ''
+          ln -s "${targetLlvmLibraries.compiler-rt.out}/lib" "$rsrc/lib"
+          ln -s "${targetLlvmLibraries.compiler-rt.out}/share" "$rsrc/share"
+        '';
+    in
+    {
+      compiler-rt = libraries.compiler-rt-libc;
```

Comment:
Do you actually need a custom / old `compiler-rt` here? My guess, as I mentioned in other comments, is that there is no need for a full wrapped Clang executable here, but if there is and it needs to link against `compiler-rt` (for GPU output? seems unlikely to me), you can probably just use our default `compiler-rt` version.